stardao-mongodb/src/test/java/io/stardog/stardao/mongodb/AbstractMongoDaoTest.java

```diff
@@ -104,6 +119,19 @@ public void testFindByQuery() throws Exception {
     }
 
     @Test
+    public void testFindByQueryPartial() throws Exception {
+        TestUser created1 = dao.create(TestUser.builder().name("Ian").email("ian1@example.com").build());
+        TestUser created2 = dao.create(TestUser.builder().name("Ian").email("ian2@example.com").build());
+
+        Results<TestUser,ObjectId> results = dao.findByQuery(new Document("name", "Ian"), new Document("email", -1), new Document("name", 1));
```

Comment:
Yes and maybe I should be more clear about that. `findByQuery()` is intended for fetching small resultsets where you know in advance that the result will fit into memory. For larger resultsets, you should use `iterateByQuery()`